### PR TITLE
[SPARK] fix merge into for delta

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
### Problem

Symptoms: `MERGE INTO` is not generating column level lineage for delta queries. 

This is connected with some other issues that should be resolved within this PR:
 * column level lineage is not tested for each supported spark version (its tests should be moved to `app` subproject)
 * column level lineage does not support including visitors per Spark version

Closes: #1919

### Solution

 * run column lineage tests on all supported versions, 
 * implement merge into per spark (per delta version)

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [ ] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [ ] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

**Comment by pawel-big-lebowski** [comment]
TODO: after merge modify file https://openlineage.io/docs/integrations/spark/spark_column_lineage -> change `META-INF/services/io.openlineage.spark3.agent.lifecycle.plan.column.CustomColumnLineageVisitor` to `META-INF/services/io.openlineage.spark.agent.lifecycle.plan.column.CustomColumnLineageVisitor`

File: integration/spark/app/src/test/java/io/openlineage/spark/agent/column/ColumnLevelLineageDeltaTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    context =
+        OpenLineageContext.builder()
+            .sparkSession(Optional.of(spark))
+            .sparkContext(spark.sparkContext())
+            .openLineage(new OpenLineage(Versions.OPEN_LINEAGE_PRODUCER_URI))
+            .queryExecution(queryExecution)
+            .build();
+
+    // todo: verify this in integration test ???
```
Comment (pawel-big-lebowski) [inline integration/spark/app/src/test/java/io/openlineage/spark/agent/column/ColumnLevelLineageDeltaTest.java:92]: aah, this got verirified. will remove `todo`

File: integration/spark/app/build.gradle [hunk 2]
```diff
@@ -184,10 +186,22 @@ def commonTestConfiguration = {
     classpath = project.sourceSets.test.runtimeClasspath
 }
 
+task nonParallelTest (type: Test) {
```
Comment (mobuchowski) [inline integration/spark/app/build.gradle:189]: Good idea 👍

**Review by mobuchowski — APPROVED** [review]
Amazing effort @pawel-big-lebowski 🚀 🔥 🎉

**Pull request merged**